### PR TITLE
server: Include rendition name in multipart header.

### PR DESCRIPTION
**Pull request by j0sh**
This saves us from having to make the ordering of rendition
results an implicit part of the multipart API contract.

**What does this pull request do? Explain your changes. (required)**
* Adds a `Rendition-Name` field to the HTTP push multipart return header. 
* Adds a `Content-Disposition: attachment; filename=` field to the return header. The filename follows the format: `<rendition-name>_<seq-num>.<ext>` where the `<ext>` is either `ts` or `txt`, depending on whether a blob or URL is being returned.

**How did you test each of these updates (required)**
Added checking for additional headers to unit tests

**Does this pull request close any open issues?**

**Checklist:**

- [ ] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: server/mediaserver.go [hunk 1]
```diff
@@ -701,6 +701,7 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 		hdrs := textproto.MIMEHeader{
 			"Content-Type":   {typ},
 			"Content-Length": {strconv.Itoa(length)},
+			"Rendition-Name": {cxn.params.profiles[i].Name},
```
Comment (iameli) [inline server/mediaserver.go]: Confirming that this'll work with `presets` as well as `profiles` in a webhook response?

**Review by iameli — COMMENTED** [review]
Any sense in populating the `Content-Disposition` header with a filename? Might make it a bit easier to use with existing multipart libraries; I was using [Apache Commons FileUpload](https://commons.apache.org/proper/commons-fileupload/) and it assigns those names automatically if that field is present. Perhaps something like

```
Content-Disposition: form-data; filename="P144p25fps16x9_173.ts" 
```

Usually we'd call that file `P144p25fps16x9/173.ts` but slashes in a filename field make me nervous.

File: server/mediaserver.go (hunk 1 above)
Comment (j0sh) [reply server/mediaserver.go]: Yes - if a name isn't specified in the webhook, then we'll create a default name. The client will then have to do a little more work to match up with the rendition, but they should really just set a profile name if they plan to use multipart.

https://github.com/livepeer/go-livepeer/blob/6a485e197c9df6acc0067e55dc80a274fe0cba77/server/mediaserver.go#L205-L211

**Comment by j0sh** [comment]
> Any sense in populating the Content-Disposition header with a filename? 
> Content-Disposition: form-data; filename="P144p25fps16x9_173.ts"
>

Good idea, the proposed scheme sounds good. We'll probably want `Content-Disposition` in addition to `Rendition-Name`. Otherwise the client has to go through another step to split the rendition name from the sequence number / extension. (And the fact we may be returning a `webhook_` prefix only makes it more exciting, because a naive split on underscores wouldn't work.)

**Comment by iameli** [comment]
Yeah, totally agreed that we'd want both.

**Comment by j0sh** [comment]
Added `Content-Disposition: attachment; filename="..."` in 92c00794 after some follow-up on Discord

**Review by iameli — APPROVED** [review]
LGTM!

**Comment by j0sh** [comment]
Updated to use a `txt` extension in the filename if we're returning a URL: 9cd1f180

**Review by darkdarkdragon — APPROVED** [review]
LGTM

**Pull request merged**